### DISCUSSION TITLE
Working with VScodium

### DISCUSSION THREAD

**Discussion by Ranguna** [community/community#19512] [Copilot Conversations] [answered]
Anyone out there able to successfully run copilot with vscodium ?

I've tried following these guides:
https://github.com/github-community/community/discussions/15026#discussioncomment-3000383
https://github.com/orgs/github-community/discussions/6629#discussioncomment-1524627

All I was able to get past the login issue, but now I'm getting this error: "GitHub Copilot requires VS Code 1.57+. Please update your VS Code" with this in the dev console:

```
CRITI Extension 'github.copilot' appears in product.json but enables LESS API proposals than the extension wants.
package.json (LOSES): inlineCompletionsNew, inlineCompletionsAdditions
product.json (WINS): 

log.ts:313   ERR [GitHub.copilot]: editor/inlineCompletions/actions is a proposed menu identifier. It requires 'package.json#enabledApiProposals: ["inlineCompletionsAdditions"]' and is only available when running out of dev or with the following command line switch: --enable-proposed-api GitHub.copilot
```

What's really strange is that the `product.json (WINS): ` line is empty, as if product.json didn't have anything for copilot, but it does have.

Anyone out there was able to figure this out ?

And before any microsoft people jump on this thread saying that vscodium is not supported and that open source software sucks, please don't, move on to your next post and just ignore this one.

**Comment by Ranguna** [comment, marked as answer]
Scratch everything I said!
I did the guide in comment https://github.com/orgs/github-community/discussions/15026#discussioncomment-3000383 from the start and it all works now!

Microsoft devs, next time you say that copilot is not supported in vscodium, I hope you remember this post and stop spreading misinformation.

**Reply by Ranguna** [reply]
It seems that the original comment was deleted. I tried to write it all in this post: https://ranguna.prose.sh/vscodium-and-copilot. Hopefully I didn't miss anything.

**Comment by davecheney** [comment]
Thank you. I want to say that I understand your frustrations. Personally I wish there was a better way to make it clearer that Copilot did not work on VSCodium, but I do want add that we do understand that there are VSCodium users out there -- and OSS Code users -- and while Copilot does not currently work out of the box on those editors, this is not a deliberate decision, its simply a side effect of a small team who have to prioritise their resources. I cannot promise you that we will support VSCodium in the future, but please understand that we hear you, and your request has not been ignored.

**Reply by Ranguna** [reply]
That's the thing, saying "Copilot did not work on VSCodium," is not true and that's spreading misinformation.
Copilot does work with vscodium, people just have to do a workaround to make it work.
Instead of saying that copilot doesn't work, you could say that some users were able to make it work with a workaround, point to said workaround, and let users know that, although you can't promise to make it work out of the box, the issue has been noted, and for now, users can try the workaround until the team has some time to investigate this better.

The vague ways that the teams says vscodium is not supported, makes it sound like it was a deliberate decision. You just have to work on your communication skills, that's all. 

Saying a blanket statement like "it doesn't work" just throws people of because that's not true.

Also, could you please make my other answer as the real answer instead of yours?
I can't seem to do it myself.
EDIT: done, seems the mobile UI doesn't have the "Mark as answer button".

**Reply by davecheney** [reply]
Two things;

1. as I wrote above, Copilot does not work with VSCodium _out of the box_.
2. As much as I appreciate your workaround, this is, and with no judgement applied, brittle. It may stop working in the future, as I believe the team has tried in that past to revive support for source builds of VScode — it goes deeper than the proposed apis — so my recommendation is to not work around the problem as this does not entice the development team to increase the priority of supporting VScodium.

**Reply by Ranguna** [reply]
> Copilot does not work with VSCodium out of the box.

Comments on other posts simply say that copilot doesn't work. Full stop.

https://github.com/orgs/github-community/discussions/15026#discussioncomment-2997247

https://github.com/orgs/github-community/discussions/18449#discussioncomment-2934464

https://github.com/orgs/github-community/discussions/15791#discussioncomment-2664967

> so my recommendation is to not work around the problem as this does not entice the development team to increase the priority of supporting VScodium.

So your are forbidding users to work their way around the issue?

Users can do whatever they want, it's thanks to the users that I'll be paying 100€/year. It's thanks to that workaround.

You have a community that temporarily fixes issues until the dev team has more time to work on the real solution, you should embrace that, not extinguish it.

Also, working around the problem has nothing to do with enticing the development team to increase priority on anything. Your team has its priorities and users respect that. In the mean time, users will gladly fix the issues by themselves and keep paying the subscription.

**Reply by Ranguna** [reply]
> As much as I appreciate your workaround, this is, and with no judgement applied, brittle. 

That's true yeah. Things can stop working from one update to the other.

**Reply by stockh0lm** [reply]
What is the status of this? I found your blog. Is your workaround still working?

**Reply by Ranguna** [reply]
Should be, give it a try. Maybe you'll have to add `console.log`s in more places than the one mentioned.
